corehq/apps/users/models.py

```diff
@@ -3041,3 +3041,26 @@ def role(self):
         elif self.domain:
             return CouchUser.from_django_user(self.user).get_domain_membership(self.domain).role
         return None
+
+
+class UserHistory(models.Model):
```

Comment:
I am working on adding report here which would add the indexes, and i expect that to get rolled out along with this PR.
https://github.com/dimagi/commcare-hq/pull/29909

Reply:
I don't see any indexes in that PR either. Unless you're still deciding what indexes to add I think it would be good to put them in this PR

Reply:
Yup, that is still WIP. Cool, i will add them here itself.

Reply:
Added index only on domain in https://github.com/dimagi/commcare-hq/pull/29887/commits/ab94e29e04ff59f834f36b9cda3ad9f2582c84f3

I tested with 1000 rows locally, to understand what indexes would get used using EXPLAIN.
Though had doubts if that is good enough data to test this.
I tried individual indexes on domain/user_id/by_user_id and also indexed together (domain, user_id), but index was being used only to filter domain.

I believe this was so because we are using `__in` in reports for user ids, so any index on user ids was not getting used. So, for now, i just added index for domain only, which will be used for reports as seen in [draft PR for reports so far](https://github.com/dimagi/commcare-hq/pull/29909/files#diff-97674072217e37379be148b9fdf743f686c009d5d9741d8ca3607fa16a4d526fR90).